src/plugins/formulas/formulas.js

```diff
@@ -1,425 +1,197 @@
-import EventManager from '../../eventManager';
+import staticRegister from '../../utils/staticRegister';
+import { HyperFormula } from 'hyperformula/dist/unoptimized-full/bundle.js';
 import { registerPlugin } from '../../plugins';
-import { isFormulaExpression, toUpperCaseFormula, isFormulaExpressionEscaped, unescapeFormulaExpression } from './utils';
-import Sheet from './sheet';
-import DataProvider from './dataProvider';
-import UndoRedoSnapshot from './undoRedoSnapshot';
+import { parseHFValue } from './utils';
+
+staticRegister('formulas').register('hyperformula', HyperFormula.buildEmpty());
+staticRegister('formulas').register('sheetMapping', new Map());
 
-/**
- * The formulas plugin.
- *
- * @plugin Formulas
- * @experimental
- */
 class Formulas extends BasePlugin {
   constructor(hotInstance) {
     super(hotInstance);
-    /**
-     * Instance of {@link EventManager}.
-     *
-     * @private
-     * @type {EventManager}
-     */
-    this.eventManager = new EventManager(this);
-    /**
-     * Instance of {@link DataProvider}.
-     *
-     * @private
-     * @type {DataProvider}
-     */
-    this.dataProvider = new DataProvider(this.hot);
-    /**
-     * Instance of {@link Sheet}.
-     *
-     * @private
-     * @type {Sheet}
-     */
-    this.sheet = new Sheet(this.hot, this.dataProvider);
-    /**
-     * Instance of {@link UndoRedoSnapshot}.
-     *
-     * @private
-     * @type {UndoRedoSnapshot}
-     */
-    this.undoRedoSnapshot = new UndoRedoSnapshot(this.sheet);
-    /**
-     * Flag which indicates if table should be re-render after sheet recalculations.
-     *
-     * @type {Boolean}
-     * @default false
-     * @private
-     */
-    this._skipRendering = false;
-  }
 
-  /**
-   * Checks if the plugin is enabled in the handsontable settings. This method is executed in {@link Hooks#beforeInit}
-   * hook and if it returns `true` than the {@link Formulas#enablePlugin} method is called.
-   *
-   * @returns {Boolean}
-   */
-  isEnabled() {
-    /* eslint-disable no-unneeded-ternary */
-    return this.hot.getSettings().formulas ? true : false;
-  }
+    // TODO: docs
+    this.formulasStaticCollection = staticRegister('formulas');
 
-  /**
-   * Enables the plugin functionality for this Handsontable instance.
-   */
-  enablePlugin() {
-    if (this.enabled) {
-      return;
-    }
-    const settings = this.hot.getSettings().formulas;
+    // TODO: docs
+    this.hyperformula = this.formulasStaticCollection.getItem('hyperformula');
 
-    if (isObject(settings)) {
-      if (isObject(settings.variables)) {
-        objectEach(settings.variables, (value, name) => this.setVariable(name, value));
-      }
-    }
+    this.hyperformula.addSheet(this.hot.guid.split('_')[1]); // TODO: the `_` character seems to not work within Sheet names.
+
+    // TODO: docs
+    this.sheetId = this.hyperformula.sheetMapping.fetch(this.hot.guid.split('_')[1]); // TODO: this should probably in the main HF API?
```

Comment:
You have duplicated `this.hot.guid.split('_')[1]` code, just sayin' 🙄 We need a helper function/getter for the worksheet name to modify the `sheetId` in a single place.

I don't think that `guid` will be the best ID. Same with the `id` attribute, they are too limited. Worksheet name can contain spaces or special characters. IMHO we should let the user name the worksheet in the plugin settings.

Reply:
> IMHO we should let the user name the worksheet in the plugin settings.

And if it's not provided, fall back to `guid`, or something else?

Reply:
If the worksheet name is not provided it will be auto-generated by HyperFormula with the localized pattern (`Arkusz 1`, `Arkusz 2`). So `this.sheetId` should be set to a name returned by `addSheet`.